crates/bevy_input/src/gamepad.rs

```diff
@@ -192,8 +192,10 @@ impl AxisSettings {
             1.0
         } else if new_value <= self.negative_high {
             -1.0
+        } else if new_value > self.positive_low {
+            (new_value - self.positive_low) / (self.positive_high - self.positive_low)
         } else {
-            new_value
+            -1.0 * (new_value - self.negative_low) / (self.negative_high - self.negative_low)
         };
 
         if let Some(old_value) = old_value {
```

Comment:
There's actually a strange edge case with the threshold value here. Since the scaling function "stretches" the unfiltered values over the entire range, smaller physical movements are increased to make larger virtual movements. While this is literally the intention of the change, this has the side effect of making the threshold value relatively smaller than before. With this change, a small wiggle in the physical axis would be magnified to a much larger wiggle after being filtered, which may end up larger than the original threshold. (This gets worse the smaller the sensitivity range is set.) This is unintuitive, since, as far as I know, devices with need large deadzones are unlikely to also need smaller noise thresholds. 

I'm not sure what the best way to handle this is. Simply scaling up the threshold value causes other problems (as it would be larger than intended for the clamped ranges). I _think_ the best approach is to operate on the unfiltered values directly, but getting the previous unfiltered value may be difficult.

Reply:
Good catch, thanks!
As of 07c41628a0eb99a49d1ba95024ce76f64e6fb25c We are now comparing unscaled values, but it's adding a bit more complexity into the code

Reply:
Another fun(?) note: this check is skipped if `old_value` is a different sign from `new_value`. However, in the vast majority of cases, this would be handled by the clamping from `positive_low` and `negative_low`. It does implicitly assume that the interior deadzone will be larger than the threshold, but I think that's totally sensible. Nice work!

Reply:
Thanks!
At first I wanted to do something like that:
```rust
        if let Some(old_value) = old_value {
            let unscale_new_value = if new_value >= 0.0 {
                unscale(new_value, self.positive_low, self.positive_high)
            } else {
                unscale(new_value, self.negative_low, self.negative_high)
            };
            let unscaled_old_value = if old_value >= 0.0 {
                unscale(old_value, self.positive_low, self.positive_high)
            } else {
                unscale(old_value, self.negative_low, self.negative_high)
            };

            if (unscale_new_value - unscaled_old_value).abs() <= self.threshold {
                return None;
            }
```
but in this case if we go from 0.0001 to -0.0001, we'll unscale to 0.05 to -0.05 which is above the threshold.
If we want to avoid any of those cases, I think we need to store the unscaled value instead of re-calculating them.

Reply:
Right, that makes sense. I think the current approach is good.